src/app/public/services/auth-service/auth.service.ts

```diff
@@ -0,0 +1,105 @@
+import { backendDomain } from './../../../app.module';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { catchError, Observable, of, tap } from 'rxjs';
+import { JwtHelperService } from '@auth0/angular-jwt';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { FormGroup } from '@angular/forms';
+import {
+  LoginRequest,
+  LoginResponse,
+  RegisterResponse,
+} from '../../interfaces';
+import { ApiService } from '../../api';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  constructor(
+    private http: HttpClient,
+    private snackbar: MatSnackBar,
+    private jwtService: JwtHelperService,
+    private apiService: ApiService
+  ) {}
+
+  login(loginRequest: LoginRequest): Observable<LoginResponse> {
+    return this.http
+      .post<LoginResponse>(`${backendDomain}auth/signin`, loginRequest)
+      .pipe(
+        catchError((error) => {
+          let errorMessage = 'Unknown error occurred';
+          if (error.status === 400) {
+            errorMessage = error.error.message;
+          } else if (error.status === 401) {
+            errorMessage = error.error.message;
+          }
+          this.snackbar.open(errorMessage, 'Close', {
+            panelClass: 'snackbar',
+            duration: 2000,
+            horizontalPosition: 'right',
+            verticalPosition: 'top',
+          });
+          return of(null as unknown as LoginResponse);
```

Comment:
If you move this logic to the interceptor level, that will let you handling all generic errors for all requests and handle them in a beautiful and user-friendly way